contract-sdk/types/src/crypto.rs

```diff
@@ -0,0 +1,77 @@
+use k256::{
+    ecdsa::{recoverable, Signature},
+    EncodedPoint,
+};
+use thiserror::Error;
+
+/// ECDSA recover error.
+#[derive(Error, Debug, cbor::Encode, cbor::Decode)]
+pub enum Error {
+    #[error("malformed input")]
+    MalformedInput,
+    #[error("malformed signature")]
+    MalformedSignature,
+    #[error("signature verification failed")]
+    VerificationFailed,
+}
+
+/// ECDSA public key recovery method.
+pub fn ecdsa_recover(input: &[u8]) -> Result<Vec<u8>, Error> {
```

Comment:
Any reason why this is part of `contract-sdk-types` crate, for tests? This pulls in the `k256` dependency into the WASM environment. Could gate it under an optional `crypto` feature and/or move it into a separate `contract-sdk-crypto` crate.

Reply:
Yes, it's so that we can reuse the code in the `MockEnv` for tests. We'll probably want to do this for all crypto imports, so maybe having it in a separate `contract-sdk-crypto` crate makes snse.

Reply:
done